src/recording/macromanager.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <QObject>
+
+#include "engine/enginemaster.h"
+#include "mixer/playermanager.h"
+#include "preferences/usersettings.h"
+#include "recording/recordingmanagerbase.h"
+
+#define MAX_MACRO_SIZE 1000
+#define NO_DECK nullptr
+
+struct MacroAction {
+    MacroAction(){};
+    MacroAction(double position, double target)
+            : position(position), target(target){};
+    double position;
+    double target;
+};
+
+struct Macro {
+    int length = 0;
+    MacroAction actions[MAX_MACRO_SIZE];
+    QString deck = NO_DECK;
+
+    Macro() {
+        qDebug() << "Constructing Macro";
+        std::fill_n(actions, MAX_MACRO_SIZE, MacroAction());
+    }
+
+    void appendHotcueJump(double origin, double target) {
+        qDebug() << "Macro: Appending jump";
+        actions[length].position = origin;
+        actions[length].target = target;
+        length++;
```

Comment:
This is unsafe. I think length needs to be an atomic. You retrieve the value and increment it in the same CPU cycle. Also you need to check if it's out of bounds.

Also, please prefix member variables with `m_`.

Reply:
As for the prefix: wasn't quite sure whether it is also mandatory for data structures.

This function is only ever called from the RT thread, how could there be a race condition here?

Reply:
Someome has to read those values, otherwise recording them would be pointless.

Reply:
See discussion below